### PR TITLE
feat(backend): Make number of persistence worker goroutine configurable

### PR DESCRIPTION
**Pull request by frozeNinK**

### CODE REVIEW

**Comment by frozeNinK** [comment]
seems that the travis build on python 3+ is missing mTLS:
```zsh
Traceback (most recent call last):
  File "taxi_pipeline_kubeflow_gcp_test.py", line 23, in <module>
    from tfx.examples.chicago_taxi_pipeline import taxi_pipeline_kubeflow_gcp
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/tfx/examples/chicago_taxi_pipeline/taxi_pipeline_kubeflow_gcp.py", line 39, in <module>
    from tfx.extensions.google_cloud_ai_platform.pusher import executor as ai_platform_pusher_executor
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/tfx/extensions/google_cloud_ai_platform/pusher/executor.py", line 25, in <module>
    from tfx.extensions.google_cloud_ai_platform import runner
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/tfx/extensions/google_cloud_ai_platform/runner.py", line 27, in <module>
    from googleapiclient import discovery
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/googleapiclient/discovery.py", line 50, in <module>
    from google.auth.transport import mtls
ImportError: cannot import name 'mtls'
travis_time:end:252762d3:start=1591210311699207957,finish=1591210315578458563,duration=3879250606,event=script
[0K[31;1mThe command "python3 taxi_pipeline_kubeflow_gcp_test.py" exited with 1.
```

File: backend/src/agent/persistence/main.go [hunk 1]
```diff
@@ -125,4 +127,5 @@ func init() {
 		"/apis/v1beta1", "The base path for the ML pipeline API server.")
 	flag.StringVar(&namespace, namespaceFlagName, "", "The namespace name used for Kubernetes informers to obtain the listers.")
 	flag.Int64Var(&ttlSecondsAfterWorkflowFinish, ttlSecondsAfterWorkflowFinishFlagName, 604800 /* 7 days */, "The TTL for Argo workflow to persist after workflow finish.")
+	flag.IntVar(&numWorker, numWorkerName, 16, "Number of worker for sync job.")
```
Comment (rmgogogo) [inline backend/src/agent/persistence/main.go:130]: Would you still mark it as 2 in this PR? Thanks.

File: backend/src/agent/persistence/main.go [hunk 2]
```diff
@@ -125,4 +127,5 @@ func init() {
 		"/apis/v1beta1", "The base path for the ML pipeline API server.")
 	flag.StringVar(&namespace, namespaceFlagName, "", "The namespace name used for Kubernetes informers to obtain the listers.")
 	flag.Int64Var(&ttlSecondsAfterWorkflowFinish, ttlSecondsAfterWorkflowFinishFlagName, 604800 /* 7 days */, "The TTL for Argo workflow to persist after workflow finish.")
+	flag.IntVar(&numWorker, numWorkerName, 2, "Number of worker for sync job.")
```
Comment (Bobgy) [inline backend/src/agent/persistence/main.go:130]: Would you mind adding an explict usage of this new argument in https://github.com/kubeflow/pipelines/blob/master/manifests/kustomize/base/pipeline/ml-pipeline-persistenceagent-deployment.yaml as both an example and a default value.

File: backend/Dockerfile.persistenceagent [hunk 3]
```diff
@@ -20,4 +20,7 @@ ENV NAMESPACE ""
 # Set Workflow TTL to 1 day. The way to use a different value for a particular Kubeflow Pipelines deployment is demonstrated in manifests/kustomize/base/pipeline/ml-pipeline-persistenceagent-deployment.yaml
 ENV TTL_SECONDS_AFTER_WORKFLOW_FINISH 86400
 
-CMD persistence_agent --logtostderr=true --namespace=${NAMESPACE} --ttlSecondsAfterWorkflowFinish=${TTL_SECONDS_AFTER_WORKFLOW_FINISH}
+# NUM_WORKER indicates now many worker goroutines
+ENV NUM_WORKER 2
```
Comment (Bobgy) [inline backend/Dockerfile.persistenceagent:24]: nit: maybe `NUM_WORKERS` if I'm not mistaken with English grammer?

File: backend/src/agent/persistence/main.go (hunk 2 above)
Comment (Bobgy) [reply backend/src/agent/persistence/main.go:130]: Thanks!

**Comment by Bobgy** [comment]
You don't need to worry about the travis build, it's been flaky and not required for merge.

**Comment by Bobgy** [comment]
Thanks!
/lgtm

File: backend/src/agent/persistence/main.go [hunk 4]
```diff
@@ -108,7 +110,7 @@ func main() {
 	go swfInformerFactory.Start(stopCh)
 	go workflowInformerFactory.Start(stopCh)
 
-	if err = controller.Run(2, stopCh); err != nil {
+	if err = controller.Run(numWorker, stopCh); err != nil {
```
Comment (jingzhang36) [inline backend/src/agent/persistence/main.go:113]: Do we need some sanity check on the value? E.g., what if we pass a negative value to controller.Run as numWorker

Comment (Bobgy) [reply backend/src/agent/persistence/main.go:113]: I think the server will just fail with an error message, then the user who configured it should look at that.

File: backend/src/agent/persistence/main.go (hunk 1 above)
Comment (rmgogogo) [reply backend/src/agent/persistence/main.go]: kindly ping to mark it as less value while manifest can change it.

Comment (Bobgy) [reply backend/src/agent/persistence/main.go]: @rmgogogo You are looking at a historical revision. Current version has updated to 2.

**Pull request merged**